Travis-CI is returning error on Python 3.8

Here is the output: https://app.travis-ci.com/github/kytos-ng/kytos/jobs/529070299

```
============================= test session starts ==============================
551platform linux -- Python 3.8.7, pytest-5.4.1, py-1.10.0, pluggy-0.13.1
552rootdir: /home/travis/build/kytos-ng/kytos, inifile: setup.cfg
553collected 320 items                                                            
554
555tests/unit/test_core/test_api_server.py ...................FF........... [ 10%]
556......                                                                   [ 11%]
557tests/unit/test_core/test_atcp_server.py ..FF                            [ 13%]
558tests/unit/test_core/test_auth.py ..........                             [ 16%]
559tests/unit/test_core/test_buffers.py ..........                          [ 19%]
560tests/unit/test_core/test_common.py .....................                [ 25%]
561tests/unit/test_core/test_connection.py ...............                  [ 30%]
562tests/unit/test_core/test_controller.py ................................ [ 40%]
563..............F.......                                                   [ 47%]
564tests/unit/test_core/test_events.py .....                                [ 49%]
565tests/unit/test_core/test_exceptions.py .....                            [ 50%]
566tests/unit/test_core/test_helpers.py ....                                [ 51%]
567tests/unit/test_core/test_interface.py ................................. [ 62%]
568.                                                                        [ 62%]
569tests/unit/test_core/test_kytosd.py ......                               [ 64%]
570tests/unit/test_core/test_link.py .................                      [ 69%]
571tests/unit/test_core/test_logs.py .............                          [ 73%]
572tests/unit/test_core/test_napp_dir_listener.py ....                      [ 75%]
573tests/unit/test_core/test_napps_base.py ..................               [ 80%]
574tests/unit/test_core/test_napps_manager.py .......................       [ 87%]
575tests/unit/test_core/test_switch.py ...............................      [ 97%]
576tests/unit/test_core/test_websocket.py .                                 [ 97%]
577tests/unit/test_lib/test_helpers.py .......                              [100%]
578
579=================================== FAILURES ===================================
580_______________________ TestAPIServer.test_shutdown_api ________________________
581
582args = (<tests.unit.test_core.test_api_server.TestAPIServer testMethod=test_shutdown_api>,)
583keywargs = {}
584
585    @wraps(func)
586    def patched(*args, **keywargs):
587>       with self.decoration_helper(patched,
588                                    args,
589                                    keywargs) as (newargs, newkeywargs):
590
591/opt/python/3.8.7/lib/python3.8/unittest/mock.py:1322: 
592_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
593/opt/python/3.8.7/lib/python3.8/contextlib.py:113: in __enter__
594    return next(self.gen)
595/opt/python/3.8.7/lib/python3.8/unittest/mock.py:1304: in decoration_helper
596    arg = exit_stack.enter_context(patching)
597/opt/python/3.8.7/lib/python3.8/contextlib.py:425: in enter_context
598    result = _cm_type.__enter__(cm)
599/opt/python/3.8.7/lib/python3.8/unittest/mock.py:1416: in __enter__
600    if spec is None and _is_async_obj(original):
601/opt/python/3.8.7/lib/python3.8/unittest/mock.py:51: in _is_async_obj
602    if hasattr(obj, 'm__func__'):
603../../../virtualenv/python3.8.7/lib/python3.8/site-packages/werkzeug/local.py:347: in __getattr__
604    return getattr(self._get_current_object(), name)
605../../../virtualenv/python3.8.7/lib/python3.8/site-packages/werkzeug/local.py:306: in _get_current_object
606    return self.__local()
607_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
608
609name = 'request'
610
611    def _lookup_req_object(name):
612        top = _request_ctx_stack.top
613        if top is None:
614>           raise RuntimeError(_request_ctx_err_msg)
615E           RuntimeError: Working outside of request context.
616E           
617E           This typically means that you attempted to use functionality that needed
618E           an active HTTP request.  Consult the documentation on testing for
619E           information about how to avoid this problem.
620
621../../../virtualenv/python3.8.7/lib/python3.8/site-packages/flask/globals.py:38: RuntimeError
622____________________ TestAPIServer.test_shutdown_api__error ____________________
623
624args = (<tests.unit.test_core.test_api_server.TestAPIServer testMethod=test_shutdown_api__error>,)
625keywargs = {}
626
627    @wraps(func)
628    def patched(*args, **keywargs):
629>       with self.decoration_helper(patched,
630                                    args,
631                                    keywargs) as (newargs, newkeywargs):
632
633/opt/python/3.8.7/lib/python3.8/unittest/mock.py:1322: 
634_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
635/opt/python/3.8.7/lib/python3.8/contextlib.py:113: in __enter__
636    return next(self.gen)
637/opt/python/3.8.7/lib/python3.8/unittest/mock.py:1304: in decoration_helper
638    arg = exit_stack.enter_context(patching)
639/opt/python/3.8.7/lib/python3.8/contextlib.py:425: in enter_context
640    result = _cm_type.__enter__(cm)
641/opt/python/3.8.7/lib/python3.8/unittest/mock.py:1416: in __enter__
642    if spec is None and _is_async_obj(original):
643/opt/python/3.8.7/lib/python3.8/unittest/mock.py:51: in _is_async_obj
644    if hasattr(obj, '__func__'):
645../../../virtualenv/python3.8.7/lib/python3.8/site-packages/werkzeug/local.py:347: in __getattr__
646    return getattr(self._get_current_object(), name)
647../../../virtualenv/python3.8.7/lib/python3.8/site-packages/werkzeug/local.py:306: in _get_current_object
648    return self.__local()
649_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
650
651name = 'request'
652
653    def _lookup_req_object(name):
654        top = _request_ctx_stack.top
655        if top is None:
656>           raise RuntimeError(_request_ctx_err_msg)
657E           RuntimeError: Working outside of request context.
658E           
659E           This typically means that you attempted to use functionality that needed
660E           an active HTTP request.  Consult the documentation on testing for
661E           information about how to avoid this problem.
662
663../../../virtualenv/python3.8.7/lib/python3.8/site-packages/flask/globals.py:38: RuntimeError
664__________________ TestKytosServerProtocol.test_data_received __________________
665
666self = <tests.unit.test_core.test_atcp_server.TestKytosServerProtocol object at 0x7f90aa5616a0>
667mock_kytos_event = <MagicMock name='KytosEvent' id='140259307180960'>
668
669    @patch('kytos.core.atcp_server.KytosEvent')
670    def test_data_received(self, mock_kytos_event):
671        """Test data_received method."""
672        buffers = self.server_protocol.server.controller.buffers
673        self.connection.protocol.name = 'protocol'
674    
675>       self.server_protocol.data_received(b'data')
676
677tests/unit/test_core/test_atcp_server.py:92: 
678_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
679kytos/core/atcp_server.py:173: in data_received
680    self._loop.create_task(self.server.controller.buffers.raw.aput(event))
681_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
682
683self = <_UnixSelectorEventLoop running=False closed=False debug=False>
684coro = <MagicMock name='mock.controller.buffers.raw.aput()' id='140259308707168'>
685
686    def create_task(self, coro, *, name=None):
687        """Schedule a coroutine object.
688    
689        Return a task object.
690        """
691        self._check_closed()
692        if self._task_factory is None:
693>           task = tasks.Task(coro, loop=self, name=name)
694E           TypeError: a coroutine was expected, got <MagicMock name='mock.controller.buffers.raw.aput()' id='140259308707168'>
695
696/opt/python/3.8.7/lib/python3.8/asyncio/base_events.py:431: TypeError
697_________________ TestKytosServerProtocol.test_connection_lost _________________
698
699self = <tests.unit.test_core.test_atcp_server.TestKytosServerProtocol object at 0x7f90aa2ffd90>
700mock_kytos_event = <MagicMock name='KytosEvent' id='140259310195760'>
701
702    @patch('kytos.core.atcp_server.KytosEvent')
703    def test_connection_lost(self, mock_kytos_event):
704        """Test connection_lost method."""
705        buffers = self.server_protocol.server.controller.buffers
706        self.connection.protocol.name = 'protocol'
707    
708>       self.server_protocol.connection_lost('exc')
709
710tests/unit/test_core/test_atcp_server.py:106: 
711_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
712kytos/core/atcp_server.py:194: in connection_lost
713    self._loop.create_task(self.server.controller.buffers.app.aput(event))
714_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
715
716self = <_UnixSelectorEventLoop running=False closed=False debug=False>
717coro = <MagicMock name='mock.controller.buffers.app.aput()' id='140259308885184'>
718
719    def create_task(self, coro, *, name=None):
720        """Schedule a coroutine object.
721    
722        Return a task object.
723        """
724        self._check_closed()
725        if self._task_factory is None:
726>           task = tasks.Task(coro, loop=self, name=name)
727E           TypeError: a coroutine was expected, got <MagicMock name='mock.controller.buffers.app.aput()' id='140259308885184'>
728
729/opt/python/3.8.7/lib/python3.8/asyncio/base_events.py:431: TypeError
730_____________________ TestController.test_start_controller _____________________
731
732args = (<MagicMock name='pre_install_napps' id='140259305915632'>, <MagicMock name='load_napps' id='140259306328416'>, <Async...ock name='msg_out_event_handler' id='140259306207984'>, <AsyncMock name='app_event_handler' id='140259306239216'>, ...)
733mock_pre_install_napps = <MagicMock name='pre_install_napps' id='140259305915632'>
734mock_load_napps = <MagicMock name='load_napps' id='140259306328416'>
735mock_raw_event_handler = <AsyncMock name='raw_event_handler' id='140259306302384'>
736
737    @staticmethod
738    @patch('kytos.core.controller.KytosServer')
739    @patch('kytos.core.controller.Controller.app_event_handler')
740    @patch('kytos.core.controller.Controller.msg_out_event_handler')
741    @patch('kytos.core.controller.Controller.msg_in_event_handler')
742    @patch('kytos.core.controller.Controller.raw_event_handler')
743    @patch('kytos.core.controller.Controller.load_napps')
744    @patch('kytos.core.controller.Controller.pre_install_napps')
745    def test_start_controller(*args):
746        """Test activate method."""
747        (mock_pre_install_napps, mock_load_napps, mock_raw_event_handler,
748         mock_msg_in_event_handler, mock_msg_out_event_handler,
749         mock_app_event_handler, _) = args
750    
751        napp = MagicMock()
752        loop = MagicMock()
753        options = KytosConfig().options['daemon']
754        options.napps_pre_installed = [napp]
755        controller = Controller(options, loop=loop)
756        controller.log = Mock()
757    
758        controller.start_controller()
759    
760        controller.server.serve_forever.assert_called()
761        calls = [call(mock_raw_event_handler.return_value),
762                 call(mock_msg_in_event_handler.return_value),
763                 call(mock_msg_out_event_handler.return_value),
764                 call(mock_app_event_handler.return_value)]
765>       loop.create_task.assert_has_calls(calls)
766
767tests/unit/test_core/test_controller.py:223: 
768_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
769
770self = <MagicMock name='mock.create_task' id='140259276426208'>
771calls = [call(<AsyncMock name='raw_event_handler()' id='140259276519504'>), call(<AsyncMock name='msg_in_event_handler()' id='...me='msg_out_event_handler()' id='140259276549376'>), call(<AsyncMock name='app_event_handler()' id='140259276561952'>)]
772any_order = False
773
774    def assert_has_calls(self, calls, any_order=False):
775        """assert the mock has been called with the specified calls.
776        The `mock_calls` list is checked for the calls.
777    
778        If `any_order` is False (the default) then the calls must be
779        sequential. There can be extra calls before or after the
780        specified calls.
781    
782        If `any_order` is True then the calls can be in any order, but
783        they must all appear in `mock_calls`."""
784        expected = [self._call_matcher(c) for c in calls]
785        cause = next((e for e in expected if isinstance(e, Exception)), None)
786        all_calls = _CallList(self._call_matcher(c) for c in self.mock_calls)
787        if not any_order:
788            if expected not in all_calls:
789                if cause is None:
790                    problem = 'Calls not found.'
791                else:
792                    problem = ('Error processing expected calls.\n'
793                               'Errors: {}').format(
794                                   [e if isinstance(e, Exception) else None
795                                    for e in expected])
796>               raise AssertionError(
797                    f'{problem}\n'
798                    f'Expected: {_CallList(calls)}'
799                    f'{self._calls_repr(prefix="Actual").rstrip(".")}'
800                ) from cause
801E               AssertionError: Calls not found.
802E               Expected: [call(<AsyncMock name='raw_event_handler()' id='140259276519504'>),
803E                call(<AsyncMock name='msg_in_event_handler()' id='140259276532704'>),
804E                call(<AsyncMock name='msg_out_event_handler()' id='140259276549376'>),
805E                call(<AsyncMock name='app_event_handler()' id='140259276561952'>)]
806E               Actual: [call(<coroutine object AsyncMockMixin._execute_mock_call at 0x7f90aa38e8c0>),
807E                call(<coroutine object AsyncMockMixin._execute_mock_call at 0x7f90aa38e440>),
808E                call(<coroutine object AsyncMockMixin._execute_mock_call at 0x7f90aa38e7c0>),
809E                call(<coroutine object AsyncMockMixin._execute_mock_call at 0x7f90aa38eec0>),
810E                call(<coroutine object Controller.start_controller.<locals>._run_api_server_thread at 0x7f90aa38e6c0>),
811E                call().add_done_callback(<function Controller.start_controller.<locals>._stop_loop at 0x7f90aa1c8310>)]
812
813/opt/python/3.8.7/lib/python3.8/unittest/mock.py:950: AssertionError
814=============================== warnings summary ===============================
815tests/unit/test_core/test_atcp_server.py::TestKytosServer::test_connection_to_server
816  /home/travis/build/kytos-ng/kytos/tests/unit/test_core/test_atcp_server.py:31: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
817    def wait_and_go():
818
819tests/unit/test_core/test_atcp_server.py::TestKytosServer::test_connection_to_server
820  /home/travis/build/kytos-ng/kytos/tests/unit/test_core/test_atcp_server.py:33: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
821    yield from asyncio.sleep(0.01, loop=self.loop)
822
823tests/unit/test_core/test_atcp_server.py::TestKytosServer::test_connection_to_server
824  /home/travis/build/kytos-ng/kytos/tests/unit/test_core/test_atcp_server.py:35: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
825    _ = yield from asyncio.open_connection(
826
827tests/unit/test_core/test_auth.py: 10 tests with warnings
828tests/unit/test_core/test_common.py: 21 tests with warnings
829tests/unit/test_core/test_controller.py: 54 tests with warnings
830tests/unit/test_core/test_interface.py: 29 tests with warnings
831tests/unit/test_core/test_link.py: 17 tests with warnings
832tests/unit/test_core/test_napps_manager.py: 23 tests with warnings
833tests/unit/test_core/test_switch.py: 31 tests with warnings
834tests/unit/test_lib/test_helpers.py: 2 tests with warnings
835  /home/travis/build/kytos-ng/kytos/kytos/core/config.py:167: UserWarning: Unknown arguments: ['tests/unit']
836    warnings.warn(f"Unknown arguments: {unknown}")
837
838tests/unit/test_core/test_auth.py: 40 tests with warnings
839tests/unit/test_core/test_buffers.py: 13 tests with warnings
840tests/unit/test_core/test_controller.py: 224 tests with warnings
841tests/unit/test_core/test_napps_manager.py: 92 tests with warnings
842tests/unit/test_core/test_switch.py: 124 tests with warnings
843tests/unit/test_lib/test_helpers.py: 8 tests with warnings
844  /home/travis/virtualenv/python3.8.7/lib/python3.8/site-packages/janus/__init__.py:39: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
845    self._async_mutex = asyncio.Lock(loop=loop)
846
847tests/unit/test_core/test_auth.py: 40 tests with warnings
848tests/unit/test_core/test_buffers.py: 13 tests with warnings
849tests/unit/test_core/test_controller.py: 224 tests with warnings
850tests/unit/test_core/test_napps_manager.py: 92 tests with warnings
851tests/unit/test_core/test_switch.py: 124 tests with warnings
852tests/unit/test_lib/test_helpers.py: 8 tests with warnings
853  /home/travis/virtualenv/python3.8.7/lib/python3.8/site-packages/janus/__init__.py:40: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
854    self._async_not_empty = asyncio.Condition(self._async_mutex, loop=loop)
855
856tests/unit/test_core/test_auth.py: 40 tests with warnings
857tests/unit/test_core/test_buffers.py: 13 tests with warnings
858tests/unit/test_core/test_controller.py: 224 tests with warnings
859tests/unit/test_core/test_napps_manager.py: 92 tests with warnings
860tests/unit/test_core/test_switch.py: 124 tests with warnings
861tests/unit/test_lib/test_helpers.py: 8 tests with warnings
862  /home/travis/virtualenv/python3.8.7/lib/python3.8/site-packages/janus/__init__.py:41: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
863    self._async_not_full = asyncio.Condition(self._async_mutex, loop=loop)
864
865tests/unit/test_core/test_auth.py: 40 tests with warnings
866tests/unit/test_core/test_buffers.py: 13 tests with warnings
867tests/unit/test_core/test_controller.py: 224 tests with warnings
868tests/unit/test_core/test_napps_manager.py: 92 tests with warnings
869tests/unit/test_core/test_switch.py: 124 tests with warnings
870tests/unit/test_lib/test_helpers.py: 8 tests with warnings
871  /home/travis/virtualenv/python3.8.7/lib/python3.8/site-packages/janus/__init__.py:42: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
872    self._finished = asyncio.Event(loop=self._loop)
873
874tests/unit/test_core/test_auth.py: 10 tests with warnings
875  /home/travis/virtualenv/python3.8.7/lib/python3.8/site-packages/jwt/api_jwt.py:77: DeprecationWarning: It is strongly recommended that you pass in a value for the "algorithms" argument when calling decode(). This argument will be mandatory in a future version.
876    warnings.warn(
877
878tests/unit/test_core/test_auth.py::TestAuth::test_02_list_users_request
879tests/unit/test_core/test_auth.py::TestAuth::test_03_create_user_request
880tests/unit/test_core/test_auth.py::TestAuth::test_03_create_user_request_error
881tests/unit/test_core/test_auth.py::TestAuth::test_04_list_user_request
882tests/unit/test_core/test_auth.py::TestAuth::test_04_list_user_request_error
883tests/unit/test_core/test_auth.py::TestAuth::test_05_update_user_request
884tests/unit/test_core/test_auth.py::TestAuth::test_05_update_user_request_error
885tests/unit/test_core/test_auth.py::TestAuth::test_06_delete_user_request
886tests/unit/test_core/test_auth.py::TestAuth::test_06_delete_user_request_error
887  /home/travis/virtualenv/python3.8.7/lib/python3.8/site-packages/jwt/api_jws.py:141: DeprecationWarning: It is strongly recommended that you pass in a value for the "algorithms" argument when calling decode(). This argument will be mandatory in a future version.
888    warnings.warn(
889
890tests/unit/test_core/test_kytosd.py::TestKytosd::test_main__background
891  /opt/python/3.8.7/lib/python3.8/inspect.py:513: RuntimeWarning: coroutine 'async_main.<locals>.start_shell_async' was never awaited
892    def _is_wrapper(f):
893
894-- Docs: https://docs.pytest.org/en/latest/warnings.html
895=========================== short test summary info ============================
896FAILED tests/unit/test_core/test_api_server.py::TestAPIServer::test_shutdown_api
897FAILED tests/unit/test_core/test_api_server.py::TestAPIServer::test_shutdown_api__error
898FAILED tests/unit/test_core/test_atcp_server.py::TestKytosServerProtocol::test_data_received
899FAILED tests/unit/test_core/test_atcp_server.py::TestKytosServerProtocol::test_connection_lost
900FAILED tests/unit/test_core/test_controller.py::TestController::test_start_controller
901================ 5 failed, 315 passed, 2214 warnings in 24.67s =================
902
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,7 @@ python:
   - "3.6"
   - "3.7"
   - "3.8"
+  - "3.9"
 install:
   - pip install --upgrade pip
   - pip install -r requirements/dev.txt

--- a/tests/unit/test_core/test_api_server.py
+++ b/tests/unit/test_core/test_api_server.py
@@ -54,22 +54,19 @@ class TestAPIServer(unittest.TestCase):
 
         mock_exit.assert_called()
 
-    @patch('kytos.core.api_server.request')
-    def test_shutdown_api(self, mock_request):
+    def test_shutdown_api(self):
         """Test shutdown_api method."""
-        mock_request.host = 'localhost:8181'
+        path = "http://localhost:8181"
+        with self.api_server.app.test_request_context(path=path):
+            self.api_server.shutdown_api()
+            self.api_server.server.stop.assert_called()
 
-        self.api_server.shutdown_api()
-
-        self.api_server.server.stop.assert_called()
-
-    @patch('kytos.core.api_server.request')
-    def test_shutdown_api__error(self, mock_request):
+    def test_shutdown_api__error(self):
         """Test shutdown_api method to error case."""
-        mock_request.host = 'any:port'
-        self.api_server.shutdown_api()
-
-        self.api_server.server.stop.assert_not_called()
+        path = "http://any:port"
+        with self.api_server.app.test_request_context(path=path):
+            self.api_server.shutdown_api()
+            self.api_server.server.stop.assert_not_called()
 
     def test_status_api(self):
         """Test status_api method."""

--- a/tests/unit/test_core/test_atcp_server.py
+++ b/tests/unit/test_core/test_atcp_server.py
@@ -28,13 +28,11 @@ class TestKytosServer:
 
     def test_connection_to_server(self):
         """Test if we really can connect to TEST_ADDRESS."""
-        @asyncio.coroutine
-        def wait_and_go():
+        async def wait_and_go():
             """Wait a little for the server to go up, then connect."""
-            yield from asyncio.sleep(0.01, loop=self.loop)
+            await asyncio.sleep(0.01, loop=self.loop)
             # reader, writer = ...
-            _ = yield from asyncio.open_connection(
-                *TEST_ADDRESS, loop=self.loop)
+            _ = await asyncio.open_connection(*TEST_ADDRESS, loop=self.loop)
 
         self.loop.run_until_complete(wait_and_go())
 

--- a/tests/unit/test_core/test_controller.py
+++ b/tests/unit/test_core/test_controller.py
@@ -216,11 +216,8 @@ class TestController(TestCase):
         controller.start_controller()
 
         controller.server.serve_forever.assert_called()
-        calls = [call(mock_raw_event_handler.return_value),
-                 call(mock_msg_in_event_handler.return_value),
-                 call(mock_msg_out_event_handler.return_value),
-                 call(mock_app_event_handler.return_value)]
-        loop.create_task.assert_has_calls(calls)
+        loop.create_task.call_count == 5
+        assert len(controller._tasks) == 5
         mock_pre_install_napps.assert_called_with([napp])
         mock_load_napps.assert_called()
 

--- a/tests/unit/test_core/test_controller.py
+++ b/tests/unit/test_core/test_controller.py
@@ -202,9 +202,7 @@ class TestController(TestCase):
     @patch('kytos.core.controller.Controller.pre_install_napps')
     def test_start_controller(*args):
         """Test activate method."""
-        (mock_pre_install_napps, mock_load_napps, mock_raw_event_handler,
-         mock_msg_in_event_handler, mock_msg_out_event_handler,
-         mock_app_event_handler, _) = args
+        (mock_pre_install_napps, mock_load_napps, _, _, _, _, _) = args
 
         napp = MagicMock()
         loop = MagicMock()
@@ -216,7 +214,7 @@ class TestController(TestCase):
         controller.start_controller()
 
         controller.server.serve_forever.assert_called()
-        loop.create_task.call_count == 5
+        assert loop.create_task.call_count == 5
         assert len(controller._tasks) == 5
         mock_pre_install_napps.assert_called_with([napp])
         mock_load_napps.assert_called()

